apps/core/declarativewebcontainer.cpp

```diff
@@ -932,6 +933,8 @@ void DeclarativeWebContainer::initialize()
         return;
     }
 
+    SettingManager::instance()->initialize();
```

Comment:
I'd assume that this very call could be inside the SettingsManager

- EmbedLiteSearchEngine.js - profile-after-change => profile loaded
- SettingManager::handleObserve handles embed:search "init"
- At line settingsmanager.cpp:168 there call to setSearchEngine I think that could changed initialize()

Surely there must be somewhere SettingManager::instance() call and that we have in the browser.cpp

With this approach we're waiting for instance that tabsmodel is loaded which is unrelated to the initialization of the settingsmanager. What do you think?

Reply:
I agree doing it here isn't ideal, but I also wonder if triggering it on the search message is appropriate if `SettingManager` isn't really about search.

Perhaps we could drop `initialize()` entirely and just add the remaining `connect` into the constructor? There's nothing else happening in the initialize that's needed elsewhere and it streamlines things a bit.

Reply:
Yes I agree, keeping current setSearchEngine call in place in SettingManager::handleObserve + adding connection [1] to the SettingsManager constructor would make this even cleaner.

[1]
```
    connect(m_searchEngineConfItem, &MGConfItem::valueChanged,
            this, &SettingManager::setSearchEngine);
```

Reply:
I've moved the `connect` into the constructor and removed the `initialize()` code. I tested it on first startup (after deleting the browser settings folder), after a reboot (following oneshot initialisation), for the top menu search shortcut, and in the general case. They also seem to work as expected, although I had to fix https://github.com/sailfishos/embedlite-components/pull/69 to get the first-run case to work.